File: MyNeighborhood/src/main/java/com/example/project/MyNeighborhood/user/service/UserService.java
package com.example.project.MyNeighborhood.user.service;

import com.example.project.MyNeighborhood.user.model.User;
import com.example.project.MyNeighborhood.user.model.UserDTO;

import java.util.UUID;

public interface UserService {
    UserDTO getUserById(String userId);

    User getUserByEmail(String email);

    void createUser(User user);

    void updateUser(UUID userId, User updatedUser);
}

File: my-neighborhood-front/src/userService.ts
import Axios from './callerService';

let getUserById = (id) => {
  return Axios.get(`users/${id}`);
};

let updateUser = (id, credentials) => {
  return Axios.put(`users/${id}`, credentials);
};

export const userService = {
  getUserById,
  updateUser
};
